backend/src/main/java/org/http/backend/controller/ScheduleController.java

```diff
@@ -37,6 +39,16 @@ public List<Integer> getBookedSeatsForSchedule(@PathVariable String scheduleId)
         return bookingService.getBookedSeatsByScheduleId(scheduleId);
     }
 
+    @GetMapping("/movie/{movieId}")
+    public ResponseEntity<List<Schedule>> findByMovieId(@PathVariable("movieId") String movieId) {
+        try {
+            List<Schedule> schedules = scheduleService.findByMovieId(movieId);
+            return ResponseEntity.ok(schedules != null ? schedules : new ArrayList<>());
```

Comment:
```suggestion
            return ResponseEntity.ok(schedules);
```
Repository does not return null so no need to check. It either returns a populated list or an empty list according to [docs](https://docs.spring.io/spring-data/mongodb/reference/repositories/null-handling.html)

The input however can be null, so should probably check that.